bin/test-storybook.js

```diff
@@ -34,32 +40,65 @@ function sanitizeURL(url) {
   return finalURL;
 }
 
-const targetURL = sanitizeURL(process.env.TARGET_URL || `http://localhost:6006`);
-
-urlExists(targetURL, function (err, exists) {
-  if (!exists) {
-    console.error(`[test-storybook] It seems that your Storybook instance is not running at: ${targetURL}. Are you sure it's running?`)
-    process.exit(1)
-  }
-
-  executeJestPlaywright()
-});
-
-function executeJestPlaywright() {
-  const fs = require('fs');
-  const path = require('path');
-
+async function executeJestPlaywright(args) {
   const jest = require('jest');
-  let argv = process.argv.slice(2);
+  let argv = args.slice(2);
 
   const jestConfigPath = fs.existsSync('test-runner-jest.config.js')
     ? 'test-runner-jest.config.js'
-    : path.resolve(__dirname, '../playwright/test-runner-jest.config.js')
+    : path.resolve(__dirname, '../playwright/test-runner-jest.config.js');
 
-  argv.push(
-    '--config',
-    jestConfigPath
-  );
+  argv.push('--config', jestConfigPath);
 
-  jest.run(argv);
+  await jest.run(argv);
+}
+
+async function checkStorybook(url) {
+  try {
+    const res = await fetch(url, { method: 'HEAD' });
+    if (res.status !== 200) throw new Error(`Unxpected status: ${res.status}`);
+  } catch (e) {
+    console.error(
+      `[test-storybook] It seems that your Storybook instance is not running at: ${url}. Are you sure it's running?`
+    );
+    process.exit(1);
+  }
 }
+
+async function fetchStoriesJson(url) {
+  const storiesJsonUrl = new URL('/stories.json', url).toString();
```

Comment:
the trailing slash will always be there because of the `sanitizeURL` method, and if the slash is present here it will remove the last path of the URL which will make it break. I tried mealdrop.vercel.app/storybook/ and it became mealdrop.vercel.app/stories.json

```suggestion
  const storiesJsonUrl = new URL('stories.json', url).toString();
```